sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala

```diff
@@ -39,20 +39,18 @@ import org.apache.spark.sql.util.CaseInsensitiveStringMap
 /**
  * A [[TableCatalog]] that translates calls to the v1 SessionCatalog.
  */
-class V2SessionCatalog(sessionState: SessionState) extends TableCatalog {
+class V2SessionCatalog(catalog: SessionCatalog, conf: SQLConf) extends TableCatalog {
```

Comment:
This change is necessary. We create `V2SessionCatalog` in `BaseSessionStateBuilder`, where there is no `SessionState` instance.